### PR TITLE
Drupal 9 compatibility fixes

### PR DESCRIPTION
**Pull request by bbenbahloul**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -6,22 +6,22 @@
...
         "guzzlehttp/guzzle": "~6.3",
         "nikic/php-parser": "~4.10.4",
-        "openeuropa/code-review": "~1.5",
-        "openeuropa/drupal-core-require-dev": "^8.8",
-        "openeuropa/task-runner": "~1.0-beta5",
-        "phpunit/phpunit": "~6.0"
+        "openeuropa/code-review": "~1.6",
+        "openeuropa/drupal-core-require-dev": "^8.9 || ^9",
+        "openeuropa/task-runner-drupal-project-symlink": "^1.0",
+        "phpunit/phpunit": "^7 || ^8"
```
Comment (claudiu-cristea) [inline composer.json]: Why not allowing also PHPUnit ^9?

**Review by claudiu-cristea — CHANGES_REQUESTED** [review]
Looks good, I have only a remark

File: composer.json (hunk 1 above)
Comment (bbenbahloul) [reply composer.json]: I followed the drupal 9 compatibility fixes on previous modules, and since we are supporting only php 7.3, drupal 9 installations running on php 7.3 only uses PHPUnit 8 (see https://www.drupal.org/node/3176567)

Comment (claudiu-cristea) [reply composer.json]: We're not supporting ONLY PHP 7.3, we're supporting >=7.3, that meaning also PHP 7.4 and 8.0, see https://jubianchi.github.io/semver-check/#/%3E%3D7.3/8.0.0.

If one wants to contribute to this module and he/she is on PHP 8, we shouldn't limit their choice. The same goes down, Drupal 8.9 still accepts PHPUnit ^6.5. So we need `^6.5 || ^7 || ^8 || ^9`

**Comment by claudiu-cristea** [comment]
If you look to https://drone.fpfis.eu/openeuropa/description_list_field/60/8, you'll find that the module is no more compatible with Drupal 8.9. Please check with @saidatom to find out how this has been fixed in a module that he worked on.

File: composer.json [hunk 2]
```diff
@@ -6,22 +6,23 @@
...
         "guzzlehttp/guzzle": "~6.3",
         "nikic/php-parser": "~4.10.4",
-        "openeuropa/code-review": "~1.5",
-        "openeuropa/drupal-core-require-dev": "^8.8",
-        "openeuropa/task-runner": "~1.0-beta5",
-        "phpunit/phpunit": "~6.0"
+        "openeuropa/code-review": "~1.6",
+        "openeuropa/drupal-core-require-dev": "^8.9 || ^9",
+        "openeuropa/task-runner-drupal-project-symlink": "^1.0",
+        "phpspec/prophecy-phpunit": "^2",
```
Comment (saidatom) [inline composer.json]:
```suggestion
        "phpspec/prophecy-phpunit": "^1 || ^2",
```

**Review by claudiu-cristea — DISMISSED** [review]

File: composer.json [hunk 3]
```diff
@@ -6,22 +6,23 @@
...
     "require-dev": {
         "composer/installers": "~1.5",
-        "drupal/core-composer-scaffold": "^8.8",
+        "drupal/core-composer-scaffold": "^8.9 || ^9",
         "drupal/config_devel": "~1.2",
         "drupal/drupal-extension": "~4.0",
         "drupal/tmgmt": "1.11",
-        "drush/drush": "~9.0@stable",
+        "drush/drush": "~10.3",
         "guzzlehttp/guzzle": "~6.3",
```
Comment (nagyad) [inline composer.json]: This can be removed as it is drupal core dependency.

File: phpunit.xml.dist [hunk 4]
```diff
@@ -10,7 +10,7 @@
     <env name="MINK_DRIVER_ARGS_WEBDRIVER" value='["${selenium.browser}", null, "${selenium.host}:${selenium.port}/wd/hub"]'/>
   </php>
   <testsuites>
-    <testsuite>
+    <testsuite name="Description list fields tests">
```
Comment (nagyad) [inline phpunit.xml.dist:13]: We add `cacheResult="false"` to `<phpunit>` too so let's do that here as well.

**Review by nagyad — CHANGES_REQUESTED** [review]

File: composer.json [hunk 5]
```diff
@@ -15,7 +15,6 @@
         "drupal/config_devel": "~1.2",
         "drupal/drupal-extension": "~4.0",
         "drupal/tmgmt": "1.11",
-        "drush/drush": "~10.3",
```
Comment (nagyad) [inline composer.json]: Sorry for not being explicit in my previous comment comment, on the line of guzzle. I meant, guzzle is drupal core dependency. Revert this change please.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — APPROVED** [review]

**Review by upchuk — APPROVED** [review]
